contribs/drt/src/main/java/org/matsim/contrib/drt/speedup/DrtTeleportedRouteCalculator.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * *********************************************************************** *
+ * project: org.matsim.*
+ * *********************************************************************** *
+ *                                                                         *
+ * copyright       : (C) 2020 by the members listed in the COPYING,        *
+ *                   LICENSE and WARRANTY file.                            *
+ * email           : info at matsim dot org                                *
+ *                                                                         *
+ * *********************************************************************** *
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *   See also COPYING, LICENSE and WARRANTY file                           *
+ *                                                                         *
+ * *********************************************************************** *
+ */
+
+package org.matsim.contrib.drt.speedup;
+
+import org.matsim.api.core.v01.Coord;
+import org.matsim.api.core.v01.network.Link;
+import org.matsim.api.core.v01.population.Route;
+import org.matsim.contrib.dvrp.passenger.PassengerRequest;
+import org.matsim.contrib.dvrp.passenger.TeleportingPassengerEngine.TeleportedRouteCalculator;
+import org.matsim.core.population.routes.GenericRouteImpl;
+import org.matsim.core.utils.geometry.CoordUtils;
+
+import com.google.common.base.Preconditions;
+
+/**
+ * @author Michal Maciejewski (michalm)
+ */
+public class DrtTeleportedRouteCalculator implements TeleportedRouteCalculator {
+	private final double averageWaitingTime;
+	private final double averageInVehicleBeelineSpeed;
+
+	DrtTeleportedRouteCalculator(double averageWaitingTime, double averageInVehicleBeelineSpeed) {
+		this.averageWaitingTime = averageWaitingTime;
+		this.averageInVehicleBeelineSpeed = averageInVehicleBeelineSpeed;
+	}
+
+	@Override
+	public Route calculateRoute(PassengerRequest request) {
+		Link startLink = request.getFromLink();
+		Link endLink = request.getToLink();
+		final Coord fromActCoord = Preconditions.checkNotNull(startLink.getCoord());
+		final Coord toActCoord = Preconditions.checkNotNull(endLink.getCoord());
+		double dist = CoordUtils.calcEuclideanDistance(fromActCoord, toActCoord);
+		Route route = new GenericRouteImpl(startLink.getId(), endLink.getId());
+		//TODO move wait time outside the route (handle it explicitly by the TeleportingPassengerEngine)
+		int travTime = (int)(averageWaitingTime + (dist / averageInVehicleBeelineSpeed));
```

Comment:
I'm not sure what moving wait time outside would mean. In any case it's good to have a travel time set in the route which is close to the actual travel time in the simulation.

Reply:
**Background**
I started throwing passenger pickup/dropoff events some time ago. Normally they are thrown on entering and leaving vehicles. In the case of teleported rides, pickups are thrown on departure and dropoffs on arrival (along with the teleportation arrival events).

Passenger pickup/dropoff events are used to build DRT-related stats like waiting etc. A nice thing is that now with the new version of drt-speed-up we can produce same types of passenger-related stats as for the simulated DRT, because we do not depend on passenger-vehicle interaction events.

**Idea (behind that TODO)**
Since `teleported_travel_time = wait_time + ride_time`, I would like to throw passenger pickup events not exactly on departure, but after the `wait_time` amount of time. This will make the DRT stats collected with the teleported-drt runs even closer to what we have for the simulated-drt runs.

Reply:
Nice idea! Some day we will look at the output stats and forget drtspeedup was involved at all ;-)